macrotypes/typecheck.rkt

```diff
@@ -530,7 +532,52 @@
                                    (combine-out . ts) (combine-out . t-s)
                                    (for-syntax (combine-out . t-expanders) . t?s)))
                   modes)]))))
+
+         ;; -------------------------------------------
+
+         ; (type-use-error <id>)
+         ; To be raised when a type is attempted to be evaluated
+         (define-syntax-rule (type-use-error tag-expr)
+           (raise (exn:fail:type:runtime
+                   (format "~a: Cannot use ~a at run time" tag-expr 'name)
+                   (current-continuation-marks))))
```

Comment:
I prefer to not use `syntax-rules` since it adds syntax taints. You can use `define-simple-macro` instead.

Reply:
:+1: